Microsoft.VisualStudio.DSC/Microsoft.VisualStudio.DSC.psm1

```diff
@@ -240,12 +269,34 @@ function Invoke-VsInstaller
 
     $installer = Start-Process -FilePath (Get-VsInstallerPath) -ArgumentList $Arguments -PassThru
     $installer.WaitForExit();
-
-    # See script block description for error code explanation
+    $basePath = Build-BasePath
+    $activeInstallerProcess = Get-Process Setup | Where-Object { $_.Path -like "$basePath*" }
     $validErrorCodes = 0,3010,862968;
-    if($installer.ExitCode -NotIn $validErrorCodes)
+
+    if($activeInstallerProcess)
     {
-        throw "Visual Studio Installer failed with error code $($installer.ExitCode) using arguments: $Arguments"
+        $processIds = $activeInstallerProcess | Select-Object -ExpandProperty Id
+        Wait-Process -Id $processIds -Timeout 3600
+
+        # There was an installer update, check for exit codes of the installer update operation and the user requested operation
+        # No need to check for initial installer process exit code as it should have failed
+        $exitCodes = $processIds | ForEach-Object {
```

Comment:
We're not using this return value, so perhaps this could be turned into a normal for each loop for readability?